### ISSUE TITLE
DevOps Center enhanced extensibility

### ISSUE THREAD

**Issue by kfidelak94** [forcedotcom/devops-center-roadmap#77]
We want to offer more and easier ability to extend the DevOps Center application, both by customer users as well as partners.

Some areas we hear requests for extensibility include:
- Work Item custom fields
- Custom buttons/actions on DevOps Center screens
- Customization of pipeline w.r.t. pre-/post- promotion actions and gates

We're interested to hear feedback of where YOU'D like to be able to further extend DevOps Center.  Please share your thoughts and use-cases in the comments here.

**Comment by jefersonchaves** [comment]
I would love to be able to plug tools to perform Static Analysis such as **sfdx scanner** or online tools such as **Snyk**, or **Codescan** so the team would be able to have more confidence that best-practices are being followed.

**Comment by fourfourfun** [comment]
I've got a huge amount of input that I'd love to share on this - less around the actual doing of work, the processing of work, but how that aligns to the flow of work in from the business and the journey that goes on before change is committed. Very much aligning with a lot of the practices within the Strategy Designer Cert.

Happy to go at length about it, but not sure writing a massive passage on here is the best way!

**Comment by FabienTaillon** [comment]
You already know my thoughts but I'll say it again.
It's sad to see you take time to create something that already exists and is called the Salesforce Platform.
Whatever you try to build, you'll never match what's already possible with layouts, AppBuilder, Dynamic forms, Dynamic Actions, Dynamic Interactions etc

I would prefer to take time to make components available to AppBuilder (like the one to pull changes from a work item) so that we could use Salesforce standard UI and benefit from all the Platform features.

Otherwise you'll keep getting questions about lacks of features compared to AppBuilder.

**Comment by RupertBarrow** [comment]
@kfidelak94 @FabienTaillon 

What Fabien is talking about echoes some of the woes I have had for years with the development of SFDX : the team was mainly addressing Salesforce final clients, but not considering the Salesforce partner companies. Of course, you are addressing requirements of consultants, admins, developers; but the team was missing the target of Salesforce Practice Managers.
They are the ones asking for extensibility and customisability as detailed by Fabien.
With that in mind, his request makes sense : go full speed ahead and componentize DevOps Center to let Salesforce Practice Managers adapt it to their own flow and to their project management tooling they already have on the platform.

**Comment by kfidelak94** [comment]
@FabienTaillon @RupertBarrow  Yep, totally hear you.  And yep, we are in fact starting to look at this.

**Comment by kvyasZensar** [comment]
Yes wanted to have more page layout customization and custom buttons.

**Comment by Damecek** [comment]
It would be great if we could enhance/alter the under the hood sfdx commands used for deployment. 
- we could start using string replace https://github.com/orgs/forcedotcom/projects/4?pane=issue&itemId=13214420
- or define that destructive changes don't throw errors

**Comment by naveensimplify** [comment]
It would be good if we order the work items with the latest one (order by created date desc) on the top. Was there any plans for the same?

If we have an option to configure list that would be great, or else for closed items can we have separate list view. So that the default view will only show ongoing items.

**Comment by nihi-lb-dk** [comment]
Hello, we have alot of work items that requires some kind of "pre"/"post" steps.  Two example that is "straight forward" are the following:

"Pre steps example":
A workitem contains a new duplicate rule.

When deploying a duplicate rule the deployment gets rejected we hit more than 100 accounts with that rule.

https://help.salesforce.com/s/articleView?id=000383122&type=1 

The solution to that in our case was to run the following script in execute anonymous before deployment(PSEUDO):

List<Account> accountDuplicates //List that will hit the duplicate rule 
Integer duplicateNumber = 0;
for (Account acc in accs){
 acc.name = acc.name + duplicateNumber;
}

update accs;

"Post steps example":
We have a new feature - the feature is a LWC component that is rendered on the account flexipage if we have a certain permission - "NewFeatuturePermission".  We want assign specific user to have this permission when the feature is released.

This can be done by running the following post scripts:

PermissionSet pr = [SELECT Id from PermissionSet where name = 'NewFeatuturePermission' limit]

//Assigning to the User
PermissionSetAssignment pm = new PermissionSetAssignment(AssigneeId='<Id of User>',PermissionSetId=pr.id);
insert pm;

**Comment by Damecek** [comment]
Or modify when the destructive changes are being run, pre or post, as of now it is always post which is not the prefered way for some metadata

**Comment by qdmistry** [comment]
Any update on when we'll be able to see custom fields added to Work Items? We use primarily Jira to track our work and would be great to have it connected. As simple as Jira ticket number that can be clicked and can go to Jira for updates. 
Ideally, I would like to integrate where once work item is completed, I can sync with Jira and which updates the repo url, work item status, work item url etc in Jira ticket as a closing process.

**Comment by adayIvey** [comment]
where can I find some details on what was shipped

**Comment by Damecek** [comment]
@adayIvey afiak some of this is implemented in the next gen devops center

**Comment by adayIvey** [comment]
@Damecek ya, i we are using Next Gen just not have not been able to find any details on how to add buttons etc.. I don't think it is Lighting pages or anything like that but I could be wrong
